pkgs/applications/science/physics/crystfel/default.nix

```diff
@@ -0,0 +1,133 @@
+{ lib, stdenv, fetchurl, gfortran, m4, hdf5, gsl, unzip, makeWrapper, meson, git, ninja, eigen, pkg-config, wrapGAppsHook, flex, bison, doxygen, opencl-headers, ncurses, msgpack, fftw, zeromq, ocl-icd, gtk3, gdk-pixbuf }:
+
+let
+  libccp4 = stdenv.mkDerivation rec {
+    name = "libccp4";
+    version = "6.5.1";
+    src = fetchurl {
+      url = "ftp://ftp.ccp4.ac.uk/opensource/${name}-${version}.tar.gz";
+      sha256 = "1rfvjliny29vy5bdi6rrjaw9hhhhh72pw536xwvqipqcjlylf2r8";
+    };
+    nativeBuildInputs = [ gfortran m4 ];
+    buildInputs = [ hdf5 gsl ];
+
+    # libccp4 tries to read syminfo.lib by looking at an environment variable, which hinders reproducibility.
+    # We hard-code this by providing a little patch and then passing the absolute path to syminfo.lib as a
+    # preprocessor flag.
+    preBuild = ''
+      makeFlagsArray+=(CFLAGS='-DNIX_PROVIDED_SYMOP_FILE=\"$out/share/syminfo.lib\"')
+    '';
+
+    patches = [ ./libccp4-use-hardcoded-syminfo-lib.patch ];
+  };
+  # This is the statically-linked, pre-built binary of mosflm since compiling it from source turned out
+  # to be herculean task.
```

Comment:
Can you give some more detail? Maybe it is in the end way easier than you thought.